### PR TITLE
[feat] Pay to enable extension

### PR DESCRIPTION
**Pull request by motorina0**
### Summary
The functionality in this PR allows the administrators of LNbits instances to ask for payment when a regular user wants to `enable` an extension. The payment will go to a wallet specified by the admin.
Once a user pays for an extension, that extension will be accessible for that user no matter if it is upgraded or the user enables/disables multiple times.

Check this comment before merging: https://github.com/lnbits/lnbits/pull/2516#discussion_r1615934256

### Screenshots
- set price for extension
  - image
- pay for extension
  - image

https://github.com/lnbits/lnbits/assets/2951406/23cd80b6-0428-45ae-8ab5-4ecfdd1677f1

Depends on: https://github.com/lnbits/lnbits/pull/2516

- [x] do not request payment for admins
- [x] do not request payment to re-enable
- [x] move activate/deactivate and enable/disable from `views` to `api`
- [x] jmeter test for all of the above functionality

### CODE REVIEW

**Comment by arcbtc** [comment]
not only will we now have an inncentive for developers to make extensions they can sell, we also have a massive incentive for people just to run LNbits and attract users. This is really amazing

File: lnbits/core/services.py [hunk 1]
```diff
@@ -636,7 +636,7 @@ def fee_reserve_total(amount_msat: int, internal: bool = False) -> int:
 
 async def send_payment_notification(wallet: Wallet, payment: Payment):
     await websocket_updater(
-        wallet.id,
```
Comment (motorina0) [inline lnbits/core/services.py:639]: valid fix unrelated to this PR

File: lnbits/core/helpers.py [hunk 2]
```diff
@@ -77,7 +77,9 @@ async def _stop_extension_background_work(ext_id) -> bool:
         stop_fn_name = next((fn for fn in stop_fns if hasattr(old_module, fn)), None)
         assert stop_fn_name, "No stop function found for '{ext.module_name}'"
 
-        await getattr(old_module, stop_fn_name)()
```
Comment (motorina0) [inline lnbits/core/helpers.py:80]: valid fix unrelated to this PR

File: lnbits/core/migrations.py [hunk 3]
```diff
@@ -513,3 +513,13 @@ async def m019_balances_view_based_on_wallets(db):
...
+
+
+async def m020_add_column_column_to_user_extensions(db):
+    """
+    Adds extra column to user extensions.
+    """
+    try:
+        await db.execute("ALTER TABLE extensions ADD COLUMN extra TEXT")
+    except OperationalError:
+        pass
```
Comment (dni) [inline lnbits/core/migrations.py:525]: if we objuscate the issue, lets at least throw a warning

File: lnbits/core/views/generic.py [hunk 4]
```diff
@@ -414,29 +390,3 @@ async def hex_to_uuid4(hex_value: str):
         raise HTTPException(
             status_code=HTTPStatus.BAD_REQUEST, detail=str(exc)
         ) from exc
-
-
-async def toggle_extension(extension_to_enable, extension_to_disable, user_id):
```
Comment (motorina0) [inline lnbits/core/views/generic.py:419]: this logic is moved to: `/api/v1/extension/{ext_id}/enable` and `/api/v1/extension/{ext_id}/disable`

File: lnbits/core/views/generic.py [hunk 5]
```diff
@@ -111,26 +102,10 @@ async def extensions_install(
...
-                    core_app_extra.register_new_ext_routes(ext)
-
-                settings.lnbits_deactivated_extensions.remove(activate)
-
-            await update_installed_extension_state(
-                ext_id=ext_id, active=activate is not None
-            )
-
-        all_ext_ids = [ext.code for ext in all_extensions]
-        inactive_extensions = await get_inactive_extensions()
```
Comment (motorina0) [inline lnbits/core/views/generic.py:133]: this logic is moved to: `/api/v1/extension/{ext_id}/activate` and `/api/v1/extension/{ext_id}/deactivate`

File: tests/api/test_generic.py [hunk 6]
```diff
@@ -50,32 +50,3 @@ async def test_get_extensions_no_user(client):
...
-#     assert response.status_code == 400, f"{response.url} {response.status_code}"
-
-
-# check GET /extensions: enable nonexistent extension, expect code 400 bad request
-@pytest.mark.asyncio
-async def test_get_extensions_enable_nonexistent_extension(client, to_user):
-    response = await client.get(
-        "extensions", params={"usr": to_user.id, "enable": "12341234"}
-    )
-    assert response.status_code == 400, f"{response.url} {response.status_code}"
```
Comment (motorina0) [inline tests/api/test_generic.py:81]: - this logic has been moved out of `/extensions` and into `/api/v1/extension`
- it is now covered by JMeter integration tests

File: .github/workflows/jmeter.yml [hunk 7]
```diff
@@ -37,6 +37,7 @@ jobs:
         run: |
           git clone https://github.com/lnbits/lnbits-extensions
           cd lnbits-extensions
+          git checkout pay_to_enable_extension_upgrades
```
Comment (motorina0) [inline .github/workflows/jmeter.yml:40]: Before merging into dev:
 - remove this line
 - merge this PR: https://github.com/lnbits/lnbits-extensions/pull/360

**Review by dni — APPROVED** [review]

**Review by arcbtc — APPROVED** [review]

**Pull request merged**